### PR TITLE
Add DHX FIPS 186-4 domain parameter validation example

### LINKED ISSUE #14369: Resolve or remove TODO 3.0 in man7/EVP_PKEY-DH.pod
There is the following TODO about missing documentation:
```
=for comment TODO(3.0): To validate domain parameters, additional values used
during generation may be required to be set into the key.
```
There is no example of domain parameter validation.

### PR DESCRIPTION
**Pull request by t8m**
Fixes #14369

##### Checklist

- [x] documentation is added or updated

### CODE REVIEW

**Review by paulidale — APPROVED** [review]
LGTM

**Comment by paulidale** [comment]
Merged to master.

File: doc/man7/EVP_PKEY-DH.pod [hunk 1]
```diff
@@ -137,22 +137,21 @@ An B<DH> key can be generated with a named safe prime group by calling:
...
     EVP_PKEY *param_key = NULL;
     EVP_PKEY_CTX *pctx = NULL;
 
-    pctx = EVP_PKEY_CTX_new_from_name(NULL, "DH", NULL);
+    pctx = EVP_PKEY_CTX_new_from_name(NULL, "DHX", NULL);
     EVP_PKEY_paramgen_init(pctx);
 
     params[0] = OSSL_PARAM_construct_uint("pbits", &pbits);
     params[1] = OSSL_PARAM_construct_uint("qbits", &qbits);
-    params[2] = OSSL_PARAM_construct_int("gindex", &gindex);
```
Comment (slontis) [inline doc/man7/EVP_PKEY-DH.pod:154]: ARRG!!!! Please dont remove things you dont understand.. This needs to be added back in!

Comment (slontis) [reply doc/man7/EVP_PKEY-DH.pod:154]: You have added an unverifiable g by doing this.

File: doc/man7/EVP_PKEY-DH.pod [hunk 2]
```diff
@@ -175,8 +174,51 @@ An B<DH> key can be generated using domain parameters by calling:
...
+    OSSL_PARAM *merged_params = NULL;
+    EVP_PKEY_CTX *ctx = NULL, *validate_ctx = NULL;
+    EVP_PKEY *complete_domp = NULL;
+
+    EVP_PKEY_todata(received_domp, OSSL_KEYMGMT_SELECT_DOMAIN_PARAMETERS,
+                    &domain_params);
+    extra_params[0] = OSSL_PARAM_construct_octet_string("seed", seed, seedlen);
+    extra_params[1] = OSSL_PARAM_construct_int("gindex", &gindex);
+    extra_params[2] = OSSL_PARAM_construct_int("pcounter", &pcounter);
+    extra_params[3] = OSSL_PARAM_construct_int("hindex", &hindex);
```
Comment (slontis) [inline doc/man7/EVP_PKEY-DH.pod:202]: This is not quite right.. it should use either gindex or hindex here..

File: doc/man7/EVP_PKEY-DH.pod [hunk 3]
```diff
@@ -175,8 +174,51 @@ An B<DH> key can be generated using domain parameters by calling:
...
+    extra_params[4] = OSSL_PARAM_construct_end();
+    merged_params = OSSL_PARAM_merge(domain_params, extra_params);
+
+    ctx = EVP_PKEY_CTX_new_from_name(NULL, "DHX", NULL);
+    EVP_PKEY_fromdata_init(ctx);
```
Comment (slontis) [inline doc/man7/EVP_PKEY-DH.pod:207]: It is probably not great that none of these check the return values (see todata above also)

File: doc/man7/EVP_PKEY-DSA.pod [hunk 4]
```diff
@@ -35,7 +35,7 @@ An B<EVP_PKEY> context can be obtained by calling:
 
     EVP_PKEY_CTX *pctx = EVP_PKEY_CTX_new_from_name(NULL, "DSA", NULL);
 
-An B<DH> domain parameters key can be generated by calling:
+A B<DSA> domain parameters can be generated by calling:
```
Comment (slontis) [inline doc/man7/EVP_PKEY-DSA.pod:38]: 'A' should be removed here..

File: doc/man7/EVP_PKEY-DSA.pod [hunk 5]
```diff
@@ -59,7 +59,7 @@ An B<DH> domain parameters key can be generated by calling:
 
     EVP_PKEY_print_params(bio_out, param_key, 0, NULL);
 
-An B<DSA> key can be generated using domain parameters by calling:
+A B<DSA> key can be generated using domain parameters by calling:
```
Comment (slontis) [inline doc/man7/EVP_PKEY-DSA.pod:62]: And here also.

**Comment by t8m** [comment]
@slontis As this was already merged, would you please either open a new issue or preferably a PR that fixes those?

File: doc/man7/EVP_PKEY-DH.pod (hunk 3 above)
Comment (t8m) [reply doc/man7/EVP_PKEY-DH.pod:207]: We do not usually add error checking to the examples in man pages as it makes the example code hard to read. But perhaps a code comment adding the example does not do proper error checking should be added.

File: doc/man7/EVP_PKEY-DSA.pod (hunk 4 above)
Comment (richsalz) [reply doc/man7/EVP_PKEY-DSA.pod:38]: Maybe replace with "The"

**Pull request closed without merging**